src/components/SignupForm/SignupForm.jsx

```diff
@@ -1,22 +1,101 @@
+    Link,
+    Tab,
+    Tabs,
+} from '@mui/material'
+import { useState } from 'react'
 import * as styleMui from './SignupForm.styled'
 
 const iconStyle = {
     color: '#69AD28',
     fontSize: '2rem',
 }
 
+const passValidation =
+    /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{6,15}$/
+
+const emailValidation = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/
+
 export default function SignupForm({ setTypeUser, typeUser }) {
     const handleChange = (event, newValue) => {
         setTypeUser(newValue)
     }
 
+    //Bắt lỗi input
+    const [email, setEmail] = useState('')
+    const [firstName, setFirstName] = useState('')
+    const [lastName, setLastName] = useState('')
+    const [password, setPassword] = useState('')
+    const [reEnterPassword, setReEnterPassword] = useState('')
+    const [emailErrorText, setEmailErrorText] = useState('')
+    const [firstNameErrorText, setFirstNameErrorText] = useState('')
+    const [lastNameErrorText, setLastNameErrorText] = useState('')
+    const [passwordErrorText, setPasswordErrorText] = useState('')
+    const [reEnterPasswordErrorText, setReEnterPasswordErrorText] = useState('')
+    const [isTrue, setIsTrue] = useState(false)
+    const [isTrueErrorText, setIsTrueErrorText] = useState('')
+
+    //Chuyển trạng thái nhìn thấy mật khẩu
+    const [eye, setEye] = useState(false)
+    const handleEye = () => {
+        setEye(!eye)
+    }
+
+    //Bắt lỗi trước khi submit
+    const onSubmit = (e) => {
+        e.preventDefault()
+
+        if (!email) {
+            setEmailErrorText('Vui lòng nhập email')
+        } else if (!email.match(emailValidation)) {
+            setEmailErrorText('Email không hợp lệ')
+        } else {
+            setEmailErrorText('')
+        }
+        if (!firstName) {
+            setFirstNameErrorText('Vui lòng nhập tên')
+        } else {
+            setFirstNameErrorText('')
+        }
+        if (!lastName) {
+            setLastNameErrorText('Vui lòng nhập họ')
+        } else {
+            setLastNameErrorText('')
+        }
+        if (!password) {
+            setPasswordErrorText('Vui lòng nhập mật khẩu')
+        } else if (!password.match(passValidation)) {
+            setPasswordErrorText(
+                'Mật khẩu phải từ 6-15 ký tự, trong đó có ít nhất một chữ số, một chữ hoa, một chữ thường và một ký tự đặc biệt'
+            )
+        } else {
+            setPasswordErrorText('')
+        }
+        if (!reEnterPassword) {
+            setReEnterPasswordErrorText('Vui lòng xác thực mật khẩu')
+        } else if (reEnterPassword !== password) {
+            setReEnterPasswordErrorText('Mật khẩu xác thực không trùng khớp')
+        } else {
+            setReEnterPasswordErrorText('')
+        }
+
+        if (!isTrue) {
+            setIsTrueErrorText(
+                'Phải chấn nhận với điều khoản trước khi đăng ký'
+            )
+        } else {
+            setIsTrueErrorText('')
+        }
+    }
```

Comment:
Look like we have a lot of condition here, so if we have to much condition you need slice that to another function, and calling that function in to this.